app/src/main/java/com/example/indoor_volleyball/Activities/LoginActivity.java

```diff
@@ -54,7 +47,18 @@ public void onClick(View v) {
                 if (username.isEmpty() || password.isEmpty()) {
                     Toast.makeText(LoginActivity.this, "Please enter something", Toast.LENGTH_SHORT).show();
                 } else {
-                    loginUser(username, password);
+                    if (havePermission) {
+                        loginUser(username, password);
+                    } else {
+                        //TODO prompt the user again if they deny location
+                        Toast.makeText(LoginActivity.this, "Indoor Volleyball cannot run without user location try again!", Toast.LENGTH_SHORT).show();
```

Comment:
Perfect, only thing to change is to put this in the strings file since the user sees it.